### PR TITLE
Add warning to use digest instead of tags to other cosign commands

### PR DESCRIPTION
**Pull request by priyawadhwa**
fixes https://github.com/sigstore/cosign/issues/2047 for 2.0

Signed-off-by: Priya Wadhwa <priya@chainguard.dev>

### CODE REVIEW

File: internal/ui/warnings.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ui
+
+const TagReferenceMessage string = `Image reference %s uses a tag, not a digest, to identify the image to sign.
+
+This can lead you to sign a different image than the intended one. Please use a
```
Comment (haydentherapper) [inline internal/ui/warnings.go:19]: Did you want to keep the indentation?

Comment (priyawadhwa) [reply internal/ui/warnings.go]: thanks! bad copy paste

**Review by haydentherapper — APPROVED** [review]

**Review by hectorj2f — APPROVED** [review]

File: cmd/cosign/cli/copy/copy.go [hunk 2]
```diff
@@ -39,6 +40,10 @@ func CopyCmd(ctx context.Context, regOpts options.RegistryOptions, srcImg, dstIm
 	if err != nil {
 		return err
 	}
+	if _, ok := srcRef.(name.Digest); !ok {
+		msg := fmt.Sprintf(ui.TagReferenceMessage, srcImg)
+		ui.Warnf(ctx, msg)
+	}
```
Comment (Vlatombe) [inline cmd/cosign/cli/copy/copy.go:46]: What would be the better recommendation to copy multi-arch images with signature if not using tag?

Comment (znewman01) [reply cmd/cosign/cli/copy/copy.go:46]: Good point. I think that `copy` and `save` may have been collateral damage here.

Comment (znewman01) [reply cmd/cosign/cli/copy/copy.go:46]: https://github.com/sigstore/cosign/pull/2799

**Pull request merged**